Source/Scene/Instanced3DModel3DTileContent.js

```diff
@@ -189,109 +202,217 @@ define([
+        var batchTableBinaryByteLength = view.getUint32(byteOffset, true);
         byteOffset += sizeOfUint32;
 
+        var gltfByteLength = view.getUint32(byteOffset, true);
         //>>includeStart('debug', pragmas.debug);
-        if ((gltfFormat !== 0) && (gltfFormat !== 1)) {
-            throw new DeveloperError('Only glTF format 0 (uri) or 1 (embedded) are supported. Format ' + gltfFormat + ' is not');
+        if (gltfByteLength === 0) {
+            throw new DeveloperError('glTF byte length is zero, i3dm must have a glTF to instance.');
         }
         //>>includeEnd('debug');
-
-        var batchTableResources = new Cesium3DTileBatchTableResources(this, instancesLength);
-        this.batchTableResources = batchTableResources;
-        var hasBatchTable = false;
-        if (batchTableByteLength > 0) {
-            hasBatchTable = true;
-            var batchTableString = getStringFromTypedArray(uint8Array, byteOffset, batchTableByteLength);
-            batchTableResources.batchTable = JSON.parse(batchTableString);
-            byteOffset += batchTableByteLength;
+        byteOffset += sizeOfUint32;
+        
+        var gltfFormat = view.getUint32(byteOffset, true);
+        //>>includeStart('debug', pragmas.debug);
+        if (gltfFormat !== 1 && gltfFormat !== 0) {
+            throw new DeveloperError('Only glTF format 0 (uri) or 1 (embedded) are supported. Format ' + gltfFormat + ' is not.');
         }
+        //>>includeEnd('debug');
+        byteOffset += sizeOfUint32;
 
-        var gltfView = new Uint8Array(arrayBuffer, byteOffset, gltfByteLength);
-        byteOffset += gltfByteLength;
-
-        // Each vertex has a longitude, latitude, and optionally batchId if there is a batch table
-        // Coordinates are in double precision, batchId is a short
-        var instanceByteLength = sizeOfFloat64 * 2 + (hasBatchTable ? sizeOfUint16 : 0);
-        var instancesByteLength = instancesLength * instanceByteLength;
-
-        var instancesView = new DataView(arrayBuffer, byteOffset, instancesByteLength);
-        byteOffset += instancesByteLength;
-
-        // Create model instance collection
-        var collectionOptions = {
-            instances : new Array(instancesLength),
-            batchTableResources : batchTableResources,
-            boundingVolume : this._tile.contentBoundingVolume.boundingVolume,
-            cull : false,
-            url : undefined,
-            headers : undefined,
-            type : RequestType.TILES3D,
-            gltf : undefined,
-            basePath : undefined
-        };
-
-        if (gltfFormat === 0) {
-            var gltfUrl = getStringFromTypedArray(gltfView);
-            collectionOptions.url = joinUrls(this._tileset.baseUrl, gltfUrl);
-        } else {
-            collectionOptions.gltf = gltfView;
-            collectionOptions.basePath = this._url;
-        }
+        if (featureTableJSONByteLength > 0) {
```

Comment:
If this is false the promises won't resolve. I say just throw an error if `featureTableJSONByteLength` is 0.